pathSum2.py

```diff
@@ -0,0 +1,53 @@
+# To find all the paths travserse the tree recursively and add to global variable when sum equals to target sum.
+#Time complexity: O(h)
+#Space Complexity: O(num_of_arrays)
```

Comment:
Time: O(n), visits each node once.
Space: O(n^2), a new list is created at each node.